python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -503,10 +508,16 @@ def get() -> "DagsterInstance":
     @public
     @staticmethod
     def local_temp(
-        tempdir: Optional[str] = None, overrides: Optional[DagsterInstanceOverrides] = None
+        tempdir: Optional[str] = None,
+        overrides: Optional[DagsterInstanceOverrides] = None,
     ) -> "DagsterInstance":
         if tempdir is None:
-            tempdir = DagsterInstance.temp_storage()
+            created_dir = TemporaryDirectory()
+            i = DagsterInstance.from_ref(
+                InstanceRef.from_dir(created_dir.name, overrides=overrides)
+            )
+            DagsterInstance._TEMP_DIRS[i] = created_dir
```

Comment:
Does this also need a lock? Apparently not given your test case but I don't understand why